### ISSUE TITLE
API inconsistencies

### ISSUE THREAD

**Issue by Risto-Stevcev** [jsverify/jsverify#256]
- `jsc.tuple` is listed as taking a variadic argument, but it actually takes an array instead
- `jsc.sum` is listed but doesn't even exist

These are just the ones I noticed

**Comment by sirsavary** [comment]
`jsc.sum` is right here https://github.com/jsverify/jsverify/blob/master/lib/arbitrary.js#L85

**Comment by Risto-Stevcev** [comment]
Ah I should've dug deeper for `jsc.sum`, it looks like the lib version on the website is outdated

**Comment by Risto-Stevcev** [comment]
@xtianjohns Why is it a non-issue?

**Comment by sirsavary** [comment]
This definitely isn't a non-issue, the docs need a lot of TLC. I attempted to introduce this library to my team recently but unfortunately, they weren't entirely able to wrap their heads around the README. JSVerify is confusing enough without the API docs not matching the implementation.

**Comment by StoneCypher** [comment]
doc patches are accepted regularly

**Comment by xtianjohns** [comment]
Sorry folks, I just misread! I took "Ah I should've dug deeper", and quickly misinterpreted that as a documentation "mis read", rather than for what it was - seems like there are still issues.

Given that we can work on doc patches as a community, do we want this issue to be a proxy for "improve the docs"? If so, perhaps a rename and some focus on specific areas of improvement would be helpful. If folks have had problems with the docs, let's lay a few out that we can correct and then submit some patches.

I'll happily help collate feedback into a PR if that's helpful!

**Comment by glebec** [comment]
@Risto-Stevcev:

> * `jsc.tuple` is listed as taking a variadic argument, but it actually takes an array instead
> * `jsc.sum` is listed but doesn't even exist

Actually the documentation is as follows:

- `tuple(arbs: (arbitrary a, arbitrary b...)): arbitrary (a, b...)`
- `sum(arbs: (arbitrary a, arbitrary b...)): arbitrary (a | b ...)`

Looking around the README, those appear not to mean variadic arguments but arrays of arguments (hence the "value name" `arbs`). The author uses parens to mean tuples which are counted as equivalent to arrays. Variadic would be as follows:

- `tuple(arbA: arbitrary a, arbB: arbitrary b...): arbitrary (a, b...)`
- `sum(arbA: arbitrary a, arbB: arbitrary b...): arbitrary (a | b ...)`

Now, my own preference is that the JS array type be represented via square brackets, which is more idiomatic. That would look like this:

- `tuple(arbs: [arbitrary a, arbitrary b...]): arbitrary (a, b...)`
- `sum(arbs: [arbitrary a, arbitrary b...]): arbitrary (a | b ...)`

I think that would be much less confusing for JS devs.

**Comment by Risto-Stevcev** [comment]
Yeah in my opinion that's very strange syntax because JS doesn't have tuples and it doesn't follow any kind of usual convention with type signatures. If it's intended for JS audiences I'd bring it closer to typescript signatures, something like:
```
sum(arbs: Array<T>): Arbitrary<T>
```
With a reminder that polymorphic types like `T` in javascript can be intersection types, like `(string | number)`
